api/Controllers/HotelsController.cs

```diff
@@ -73,20 +73,42 @@ public async Task<IActionResult> GetHotel([FromRoute] int hotelId)
         }
 
         [HttpGet]
-        public async Task<IActionResult> GetHotelCards([FromQuery] PageParameters pageParameters)
+        public async Task<IActionResult> GetHotelCards(
+            [FromQuery] PageParameters pageParameters, 
+            [FromQuery] HotelFilterParameters filterParams
+        )
         {
-            HotelCard[] hotelCards = await _hotelsDb.Hotels
+            var filteredHotelCards = _hotelsDb.Hotels.AsQueryable();
+            if (!string.IsNullOrEmpty(filterParams.Name))
+            {
+                filteredHotelCards = filteredHotelCards.Where(h => h.Name.Contains(filterParams.Name));
+            }
+            var hotelCount = await filteredHotelCards.CountAsync();
+            HotelCard[] hotelCards = await filteredHotelCards
                 .Skip(pageParameters.PageIndex * pageParameters.PageSize)
                 .Take(pageParameters.PageSize)
                 .ProjectTo<HotelCard>(_mapper.ConfigurationProvider)
                 .ToArrayAsync();
-            return Ok(hotelCards);
+            return Ok(new { hotelCards, hotelCount });
         }
 
-        [Route("count")]
+        [Route("names")]
         [HttpGet]
-        public async Task<IActionResult> GetHotelsCount() =>
-            Ok(await _hotelsDb.Hotels.CountAsync());
+        public async Task<IActionResult> GetHotelNames([FromQuery] string name, [FromQuery] int number = 2)
+        {
+            if (string.IsNullOrEmpty(name))
+            {
+                return NoContent();
+            }
```

Comment:
For endpoints that are supposed to return a list a records I would return empty array instead of NoContent if no records was found. That will simplify processing on the frontend. Also in that case we can change return type of the action from IActionResult to actual string[].